pandas/core/indexes/multi.py

```diff
@@ -431,6 +437,70 @@ def from_product(cls, iterables, sortorder=None, names=None):
         codes = cartesian_product(codes)
         return MultiIndex(levels, codes, sortorder=sortorder, names=names)
 
+    @classmethod
+    def from_frame(cls, df, sortorder=None, names=None):
+        """
+        Make a MultiIndex from a DataFrame.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            DataFrame to be converted to MultiIndex.
+        sortorder : int or None
+            Level of sortedness (must be lexicographically sorted by that
+            level).
+        names : list-like, optonal
+            If no names are provided, use the column names, or tuple of column
+            names if the columns is a MultiIndex. If a sequence, overwrite
+            names with the given sequence.
+
+        Returns
+        -------
+        MultiIndex or Index
+            The MultiIndex representation of the given DataFrame.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame([['ahc', 'happy'], ['ahc', 'jolly'],
+        ...                    ['boh', 'happy'], ['boh', 'jolly'],
+        ...                    ['oil', 'joy'], ['oil', 'joy']],
+        ...                   columns=['a', 'b'])
+        >>> df
+             a      b
+        0  ahc  happy
+        1  ahc  jolly
+        2  boh  happy
+        3  boh  jolly
+        4  oil    joy
+        5  oil    joy
+
+        >>> pd.MultiIndex.from_frame(df)
+        MultiIndex(levels=[['ahc', 'boh', 'oil'], ['happy', 'jolly', 'joy']],
+                   labels=[[0, 0, 1, 1, 2, 2], [0, 1, 0, 1, 2, 2]],
+                   names=['a', 'b'])
+
+        # Use explicit names, instead of column names
+        >>> pd.MultiIndex.from_frame(df, names=['X', 'Y'])
+        MultiIndex(levels=[['ahc', 'boh', 'oil'], ['happy', 'jolly', 'joy']],
+                   labels=[[0, 0, 1, 1, 2, 2], [0, 1, 0, 1, 2, 2]],
+                   names=['X', 'Y'])
+
+        See Also
```

Comment:
This goes before the Examples section. `./scripts/validate_docstrings.py pandas.MultiIndex.from_frame` should report it as an error. Make sure nothing is reported by the script.

Reply:
Fixed this docstring and all the other constructor methods docstrings (since I had to modify them to update the See Alsos). I also fixed the pd.MultiIndex docstring to the best of my abilities (since I had to make some small modifications to that too). However, there were still some issues with the MultiIndex docstring:

    2 Errors found:
        Parameters {_set_identity, name, dtype} not documented
        Unknown parameters {labels}

Any ideas on how I should address these? Looks like labels is there to serve as a deprecation reminder.